packages/mini-browser/src/browser/mini-browser.ts

```diff
@@ -0,0 +1,598 @@
+
+        const transparentOverlay = document.createElement('div');
+        transparentOverlay.classList.add(MiniBrowser.Styles.TRANSPARENT_OVERLAY);
+        transparentOverlay.style.display = 'none';
+
+        const pdfContainer = document.createElement('div');
+        pdfContainer.classList.add(MiniBrowser.Styles.PDF_CONTAINER);
+        pdfContainer.id = `${this.id}-pdf-container`;
+        pdfContainer.style.display = 'none';
+
+        contentArea.appendChild(transparentOverlay);
+        contentArea.appendChild(pdfContainer);
+        contentArea.appendChild(loadIndicator);
+        contentArea.appendChild(frame);
+
+        parent.appendChild(contentArea);
+        return Object.assign(contentArea, { frame, loadIndicator, pdfContainer, transparentOverlay });
+    }
+
+    protected createIFrame(): HTMLIFrameElement {
+        const frame = document.createElement('iframe');
+        const sandbox = (this.props.sandbox || MiniBrowserProps.SandboxOptions.DEFAULT).map(name => MiniBrowserProps.SandboxOptions[name]);
+        frame.sandbox.add(...sandbox);
+        this.toDispose.push(addEventListener(frame, 'load', this.onFrameLoad.bind(this)));
+        return frame;
+    }
+
+    protected onFrameLoad(): void {
+        this.hideLoadIndicator();
+        this.focus();
+    }
+
+    protected focus(): void {
+        const contentDocument = this.contentDocument();
+        if (contentDocument !== null) {
+            contentDocument.body.focus();
+        } else {
+            this.input.focus();
+        }
+    }
+
+    protected showLoadIndicator(): void {
+        this.loadIndicator.style.display = 'block';
+    }
+
+    protected hideLoadIndicator(): void {
+        this.loadIndicator.style.display = 'none';
+    }
+
+    protected handleBack(): void {
+        if (history.length - this.initialHistoryLength > 0) {
+            history.back();
+        }
+    }
+
+    protected handleForward(): void {
+        if (history.length > this.initialHistoryLength) {
+            history.forward();
+        }
+    }
+
+    protected handleRefresh(): void {
+        history.go();
+    }
+
+    protected handleOpen(): void {
+        const location = this.frameSrc() || this.input.value;
+        if (location) {
+            this.windowService.openNewWindow(location);
+        }
+    }
+
+    protected createInput(parent: HTMLElement): HTMLInputElement {
+        const input = document.createElement('input');
+        input.type = 'text';
+        this.toDispose.pushAll([
+            addEventListener(input, 'keydown', this.handleInputChange.bind(this)),
+            addEventListener(input, 'click', () => {
+                if (input.value) {
+                    input.select();
+                }
+            })
+        ]);
+        parent.appendChild(input);
+        return input;
+    }
+
+    protected handleInputChange(e: KeyboardEvent): void {
+        const { key } = KeyCode.createKeyCode(e);
+        // TODO numeric pad enter?
```

Comment:
it's the very same keyCode for numpad in browsers. just verified. cf. https://stackoverflow.com/a/5418541/1783092